counterpartylib/lib/backend/addrindex.py

```diff
@@ -91,9 +91,20 @@ def make_call(chunk):
             executor.submit(make_call, chunk)
     return list(responses)
 
+
 def extract_addresses(txhash_list):
     logger.debug('extract_addresses, txs: %d' % (len(txhash_list), ))
     tx_hashes_tx = getrawtransaction_batch(txhash_list, verbose=True)
+
+    return extract_addresses_from_txlist(tx_hashes_tx, getrawtransaction_batch)
+
+
+def extract_addresses_from_txlist(tx_hashes_tx, _getrawtransaction_batch):
```

Comment:
or rather, just have one method here that takes a default arg, e.g. `_getrawtransaction_batch=getrawtransaction_batch` ...still a bit warty but better than two separate methods I think... keeping the docstring you have